client/src/components/ProtectedRoute/index.tsx

```diff
@@ -0,0 +1,20 @@
+import { CircularProgress } from '@material-ui/core';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+
+interface PrivateRouteProps extends RouteProps {
+  // tslint:disable-next-line:no-any
+  component: any;
+}
+
+const ProtectedRoute = ({ component: Component, children, ...rest }: PrivateRouteProps) => {
+  const { loggedInUser } = useAuth();
+  //if loading
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    return <Redirect to="/login" />;
+  }
+  return Component ? <Route {...rest} render={(props) => <Component {...props} />} /> : <>children</>;
```

Comment:
Is there ever a case where we need to render just `children`? Why not just do `Component && <Route . . . />`

Reply:
If we want to render childrens someday. I dont see a usecase for now but, for further implementation it is good to have already scalable. 
```
 <ProtectedRoute>
  <Route exact path="/signup" component={Signup} />
  <Route exact path="/signup" component={Signup} />
</ProtectedRoute>
```